plugins/Types/ClassDiagram/edu.toronto.cs.se.modelepedia.classdiagram_mavo.operator/src/edu/toronto/cs/se/modelepedia/classdiagram/operator/CDSlice.java

```diff
@@ -26,93 +27,124 @@
 
 public class CDSlice extends Slice {
 
-	// Adds impacted model elements reachable from a single model element
 	@Override
-	public void addImpactedModelElems(EObject elem, Set<EObject> impacted) {
+	// Get all model elements that are potentially impacted by the input set.
+	public Set<EObject> getAllImpactedElements(Set<EObject> elemSet) {
```

Comment:
This function, as it is, is not specific to CD and should be moved to the superclass.